### DISCUSSION TITLE
%%timeit output is not showing up in the generated markdown

### DISCUSSION THREAD

**Discussion by fm75** [executablebooks/meta#182] [Q&A]
As part of a book I want to use cells like:
> %%timeit
do_something()

and see the output in the markdown generated from 
> !jupytext notebook.ipynb --to myst

Unfortunately, the output from such cells does not seem to get into the generated markdown.
> jupytext --version
1.6.0
> jb --version
Jupyter Book: 0.8.3
MyST-NB: 0.10.1
Sphinx Book Theme: 0.0.36
MyST-Parser: 0.12.10
Jupyter-Cache: 0.4.1
NbClient: 0.5.1

I tried 
> {glue:text}

with triple backticks around the content in a code cell. Is there something I can do to capture the content automatically? I don't want to have to copy/paste the output into a markdown cell, of course.

**Comment by choldgraf** [comment]
hmmm - I think this is probably a question for `jupytext`, since that's what we're using to do the conversion. Have you asked around over there? Does `timeit` output work for other kinds of output formats?

**Reply by fm75** [reply]
I asked out it [in an existing jupytext issue](https://github.com/mwouts/jupytext/issues/643)

**Comment by akhmerov** [comment]
Could it be because standard output has no mime types and therefore can't be `glue`ed? `%%timeit` outputs to stdout, which should be shown directly under the cell, but it's impossible to glue.

**Reply by fm75** [reply]
It is shown directly under the cell. I only resorted to attempting to glue because it was not getting put into the markdown. I guess next I should try looking at the notebook JSON to see if there is something that distinguishes this output from normal output cells.

**Reply by akhmerov** [reply]
Ah, I misunderstood, my bad—I thought you're trying to build a jupyterbook. I think it's a general feature of jupytext: it won't store outputs in the text formats. AFAIK jupytext markdown doesn't have those neither.

**Reply by fm75** [reply]
Simple experiment - 2 code cells
- 1
- %%timetit
   1
 
### Looked at the emitted JSON 
by Open with JSON viewer

### Summary of the two cells:
#### Cell 1
cell_type:"code"
execution_count:1
metadata:
0:"1"
execution_count:1
metadata:
output_type:"execute_result"
0:"1"

#### Cell 2
cell_type:"code"
execution_count:2
metadata:
name:"stdout"
output_type:"stream"
0:"6.95 ns ± 0.0249 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each) "
0:"%%timeit "
1:"1"

---
1. Does the community want the data in the "name:"stdout", output_type:"stream"  output cell to be rendered in the markdown output?
2. Is this a bug in jupytext? 
3. `jupyter nbconvert --to markdown Untitled.ipynb` contains the output

**Reply by choldgraf** [reply]
I'm not sure if it's a bug or not, but definitely seems worth opening up in Jupytext as the fix to support this will be there, I believe 👍

**Reply by fm75** [reply]
> Ah, I misunderstood, my bad—I thought you're trying to build a jupyterbook. I think it's a general feature of jupytext: it won't store outputs in the text formats. AFAIK jupytext markdown doesn't have those neither.

You did not misunderstand. I am trying to build a jupyter book. I was surprised that the output did not appear in the generated markdown. I have not used jupytext for anything except for this purpose - converting a notebook to markdown. So my knowledge of it is minimal.

**Comment by chrisjsewell** [comment]
I really confused by this whole conversation lol. I think you are misunderstanding the role of jupytext.

If you run `jupytext notebook.ipynb --to myst` you won't see **any** outputs in the Markdown (from `%%timeit` or otherwise).
That's not a bug, it is just not the goal of jupytext; it is a bi-directional converter between ipynb and other formats, and also allows for pairing of multiple formats.

`jupyter nbconvert --to markdown Untitled.ipynb` is something completely different, it is a uni-directional converter, i.e. it may show you the outputs but you cannot convert back to the notebook from the markdown.

jupyter-book is essentially a more featureful/complex alternative to nbconvert, allowing for e.g. compilation and cross-linking of multiple notebooks into a single book. Again though it is really just intended for uni-directional use.

If you want a format that shows both the inputs and output in a more easily edited format, I would actually suggest converting the ipynb (which is just a JSON file) to/from YAML.
It would be quite easy to get jupyter-book to read these (see https://jupyterbook.org/file-types/jupytext.html#file-types-custom)

**Reply by choldgraf** [reply]
Ah you are right, i think I mistakenly pointed to jupytext as the culprit so that's my bad. I forgot that outputs aren't handled at all by jupytext.

So I guess the question then is going with @fm75 s suggestion of figuring out where the output of %%timeit is stored when a cell is run, and making sure that myst-nb knows how to grab it, right? I think that @fm75 originally created this issue because his outputs from timeit weren't showing up in the book

**Reply by chrisjsewell** [reply]
> because his outputs from timeit weren't showing up in the book

are you sure that is actually correct, because that is not what the initial comment suggests. It says they do not show in the markdown not the book

**Reply by choldgraf** [reply]
Nope I'm not positive - i wanted to post my assumption so that @fm75 can confirm or not

**Comment by mwouts** [comment]
Hi @fm75 , thanks for raising this. As @chrisjsewell explained above, all the Jupytext formats only store the input cells, not the outputs. The outputs can be recreated from the inputs, either by Jupyter Book, or explicitely by calling e.g. `jupytext --to ipynb --execute`, or they can also be preserved in a _paired_ `.ipynb` file. 

Note that if you are simply looking for a way to convert your `.ipynb` document to a `.md` file that would also contain the outputs, maybe you could just use `jupyter nbconvert` instead?
